packages/shared/src/components/multiLevelMenu/MultiLevelMenu.tsx

```diff
@@ -1,20 +1,38 @@
-import React, { useState, ReactElement } from 'react';
+import React, { useState, ReactElement, useEffect } from 'react';
 import { MenuItem } from '../filters/common';
 import MultiLevelMenuDetail from './MultiLevelMenuDetail';
 import MultiLevelMenuMaster from './MultiLevelMenuMaster';
 
+interface MultiLevelMenuProps {
+  menuItems: MenuItem[];
+  directlyOpenedTab?: string;
+}
+
 export default function MultiLevelMenu({
   menuItems,
-}: {
-  menuItems: MenuItem[];
-}): ReactElement {
+  directlyOpenedTab,
+}: MultiLevelMenuProps): ReactElement {
   const [multiLevelMenuDetailItem, setMultiLevelMenuDetailItem] =
     useState(null);
 
   const setMultiLevelMenuDetail = (multiLevelMenuDetail) => {
     setMultiLevelMenuDetailItem(multiLevelMenuDetail);
   };
 
+  useEffect(() => {
+    if (!directlyOpenedTab) {
+      return;
+    }
+
+    const menuItem = menuItems.find((item) => item.title === directlyOpenedTab);
+
+    if (!menuItem) {
+      return;
+    }
+
+    setMultiLevelMenuDetailItem(menuItem);
+  }, [directlyOpenedTab]);
```

Comment:
Instead of `useEffect`, we can inject the initial value for the `multiLevelMenuDetailItem` in its `useState`.
I think it's the preferred implementation

Reply:
Oh, sorry about merging early. I agree that would make things much cleaner. Let me push this one quickly.